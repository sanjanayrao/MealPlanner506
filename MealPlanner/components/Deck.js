import React from 'react';
import { View, TouchableWithoutFeedback, Dimensions, TextInput, StyleSheet, ScrollView, FlatList, AsyncStorage} from 'react-native';
import {Picker} from '@react-native-picker/picker';
import NumericInput from 'react-native-numeric-input'

import { Card, ListItem, Icon, Text } from 'react-native-elements';
import Button from './Button';
import * as controller from '../backend/controller'



export default class Meals extends React.Component{
    constructor(){
        super();
        this.state = {
            meals: [],
            servings: 4,
            user: ''
        }
    }

    async get_meals(user){
        // use controller thing
        var response = {};

        if(!user){
            user = this.state.user
        }
        await controller.get_meals(user)
        .then(function(result) {
            response = result;
        });
        
        // overwrite test meals?
        if(response.success){
            this.setState({meals: response.meals});
            //testMeals = response.meals;
        }

    }
    _retrieveData = async () => {
        try {
          const value = await AsyncStorage.getItem('user');
          if (value !== null) {
            // We have data!!
            this.setState({user: value})
            this.get_meals(value)

          }
        } catch (error) {
          // Error retrieving data
        }
      };
    componentDidMount(){
        this._retrieveData()
    }
    async generate(){
        console.log("GENERATE");
        //  GET GENERATED MEALS FROM API using the number of servings in the stateS
        var response = {};
        await controller.generate_deck(this.state.user, this.state.servings)
        .then(function(result) {
            response = result
        })

        if(response.success) {
            this.setState({meals: response.deck})
        }   
    }

    getMealCards(){
        let cards = [];

        for(const i in this.state.meals){
            cards.push(
            <Card key={i}>
                <Card.Title>
                    {this.state.meals[i].name}
                </Card.Title>
                <Text>
                    Serves: {this.state.meals[i].servings}
                </Text>
               
                <Button
                    text={'View Meal'}
                    textStyle={{color: 'white'}}
                    buttonStyle={styles.mealButton}
                    onPress={()=>this.props.navigation.navigate('Meal View', {'meal' : this.state.meals[i] } )}
                    />            
            </Card>)
        }
        return cards;
    }
    
    picker(itemValue){
        this.setState({servings: itemValue})
    }
    render(){
        return(
           <ScrollView>
               <Text style={styles.header} h2 >Meals on Deck</Text>
               <Text style={styles.header}h5>Select the number of meals you'd like to randomly generate</Text>
               <View  style={{
                                flexDirection: 'row',
                                margin: 10,
                            }}> 
                    
                    <NumericInput minValue={4} maxValue={80} value={this.state.servings} onChange={value => this.picker(value)} />

                  
                    <Button text={'GENERATE'} textStyle={{color:'white'}} buttonStyle={styles.generate} onPress={()=>this.generate()}/>
               </View>
                <View>
                    {this.getMealCards()}
                </View>
           </ScrollView>
        )
    }

}
const styles = StyleSheet.create({
    mealButton:{
      backgroundColor: '#9FC9AE', 
      padding: 10, 
      borderRadius: 10,
      height: 60,
      width: "75%",
      alignSelf: 'center',
      alignItems: 'center',
      margin:10,
      justifyContent: 'center',
      alignContent: 'center'
  
    },
    list:{
        width: '100%'

    },
    header:{
        margin: 15
      },
      generate:{
        backgroundColor: '#553555', 
        padding: 10, 
        borderRadius: 50,
        alignSelf: 'flex-end',
        alignItems: 'center',
        margin:10,
        marginTop: 25,
        marginLeft: 140
      }
  });