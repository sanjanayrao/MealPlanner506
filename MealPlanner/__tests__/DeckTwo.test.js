import React from 'react';
import {cleanup, fireEvent} from '@testing-library/react';
import Deck from '../components/Deck';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, render  } from "enzyme";
import Button from '../components/Button.ios';





configure({ adapter: new Adapter() });

it("Deck Renders Same as Prior Test", () => {
    const wrapper = render(<Deck />);
    expect(wrapper).toMatchSnapshot();
  });