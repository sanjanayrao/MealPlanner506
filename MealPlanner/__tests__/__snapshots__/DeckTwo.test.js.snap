// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck Renders Same as Prior Test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "style": "flex-direction:row;margin:10px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "margin:15px",
                },
                "children": Array [
                  Object {
                    "data": "Meals on Deck",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "text",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "style": "height:5px;width:50px;margin-top:-68px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                        "selectedindex": "0",
                        "style": "0:[object Object]",
                      },
                      "children": Array [],
                      "name": "rctpicker",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "view",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "focusable": "true",
                      "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "style": "color:white;font-size:36px",
                        },
                        "children": Array [
                          Object {
                            "data": "↺",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "text",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "view",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "focusable": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "focusable": undefined,
                      "style": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "style": "height:5px;width:50px;margin-top:-68px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                      "selectedindex": "0",
                      "style": "0:[object Object]",
                    },
                    "children": Array [],
                    "name": "rctpicker",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "items": undefined,
                      "selectedindex": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "items": undefined,
                      "selectedindex": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "view",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "focusable": "true",
                    "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "style": "color:white;font-size:36px",
                      },
                      "children": Array [
                        Object {
                          "data": "↺",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "text",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "view",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "focusable": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "focusable": undefined,
                    "style": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "style": "margin:15px",
                  },
                  "children": Array [
                    Object {
                      "data": "Meals on Deck",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "text",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "focusable": "true",
                  "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "style": "color:white;font-size:36px",
                    },
                    "children": Array [
                      Object {
                        "data": "↺",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "text",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                ],
                "name": "view",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "style": "height:5px;width:50px;margin-top:-68px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                        "selectedindex": "0",
                        "style": "0:[object Object]",
                      },
                      "children": Array [],
                      "name": "rctpicker",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "view",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "style": "margin:15px",
                    },
                    "children": Array [
                      Object {
                        "data": "Meals on Deck",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "text",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "focusable": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "focusable": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "view",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "flex-direction:row;margin:10px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "margin:15px",
                  },
                  "children": Array [
                    Object {
                      "data": "Meals on Deck",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "text",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "style": "height:5px;width:50px;margin-top:-68px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                          "selectedindex": "0",
                          "style": "0:[object Object]",
                        },
                        "children": Array [],
                        "name": "rctpicker",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "items": undefined,
                          "selectedindex": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "items": undefined,
                          "selectedindex": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "view",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "focusable": "true",
                        "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "style": "color:white;font-size:36px",
                          },
                          "children": Array [
                            Object {
                              "data": "↺",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "text",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "style": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "style": undefined,
                          },
                        },
                      ],
                      "name": "view",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "focusable": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "focusable": undefined,
                        "style": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "style": "height:5px;width:50px;margin-top:-68px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                        "selectedindex": "0",
                        "style": "0:[object Object]",
                      },
                      "children": Array [],
                      "name": "rctpicker",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "items": undefined,
                        "selectedindex": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "view",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "focusable": "true",
                      "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "style": "color:white;font-size:36px",
                        },
                        "children": Array [
                          Object {
                            "data": "↺",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "text",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "view",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "focusable": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "focusable": undefined,
                      "style": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "style": "margin:15px",
                    },
                    "children": Array [
                      Object {
                        "data": "Meals on Deck",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "text",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "focusable": "true",
                    "style": "background-color:#553555;padding:10px;border-radius:50px;width:15%;align-self:flex-end;align-items:center;margin:10px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "style": "color:white;font-size:36px",
                      },
                      "children": Array [
                        Object {
                          "data": "↺",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "text",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "view",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "style": "height:5px;width:50px;margin-top:-68px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "items": "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",
                          "selectedindex": "0",
                          "style": "0:[object Object]",
                        },
                        "children": Array [],
                        "name": "rctpicker",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "items": undefined,
                          "selectedindex": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "items": undefined,
                          "selectedindex": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "view",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "style": "margin:15px",
                      },
                      "children": Array [
                        Object {
                          "data": "Meals on Deck",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "text",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "style": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "focusable": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "focusable": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "view",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "rctscrollview",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
